code/modules/NTNet/network.dm

```diff
@@ -1,21 +1,251 @@
+/*
+ * # /datum/ntnet
+ *
+ * This class defines each network of the world.  Each root network is accessible by any device
+ * on the same network but NOT accessible to any other "root" networks.  All normal devices only have
+ * one network and one network_id.
+ *
+ * This thing replaces radio.  Think of wifi but better, bigger and bolder!  The idea is that any device
+ * on a network can reach any other device on that same network if it knows the hardware_id.  You can also
+ * search or broadcast to devices if you know what branch you wish.  That is to say you can broadcast to all
+ * devices on "SS13.ATMOS.SCRUBBERS" to change the settings of all the scrubbers on the station or to
+ * "SS13.AREA.FRED_HOME.SCRUBBERS" to all the scrubbers at one area.  However devices CANNOT communicate cross
+ * networks normality.
+ *
+ */
+
 /datum/ntnet
-	var/network_id = "Network"
-	var/list/connected_interfaces_by_id = list()		//id = datum/component/ntnet_interface
+	/// The full network name for this network ex. SS13.ATMOS.SCRUBBERS
+	var/network_id
+	/// The network name part of this leaf ex ATMOS
+	var/network_node_id
+	/// All devices on this network.  ALL devices on this network, not just this branch.
+	/// This list is shared between all leaf networks so we don't have to keep going to the
+	/// parents on lookups.  It is an associated list of hardware_id AND tag_id's
+	var/list/root_devices
+	/// This lists has all the networks in this node.  Each name is fully qualified
+	/// ie. SS13.ATMOS.SCRUBBERS, SS13.ATMOS.VENTS, etc
+	var/list/networks
+	/// All the devices on this branch of the network
+	var/list/linked_devices
+	/// Network children.  Associated list using the network_node_id of the child as the key
+	var/list/children
+	/// Parrnt of the network.  If this is null, we are a oot network
+	var/datum/ntnet/parent
+
+
+/*
+ * Creates a new network
+ *
+ * Used for /datum/controller/subsystem/networks/proc/create_network so do not
+ * call yourself as new doesn't do any checking itself
+ *
+ * Arguments:
+ * * net_id - Fully qualified network id for this network
+ * * net_part_id - sub part of a network if this is a child of P
+ * * P - Parent network, this will be attached to that network.
+ */
+/datum/ntnet/New(net_id, net_part_id, datum/ntnet/P = null)
+	linked_devices = list()
+	children = list()
+	network_id = net_id
+
+	if(P)
+		network_node_id = net_part_id
+		parent = P
+		parent.children[network_node_id] = src
+		root_devices = parent.root_devices
+		networks = parent.networks
+	else
+		network_node_id = net_id
+		parent = null
+		networks = list()
+		root_devices = list()
+		SSnetworks.root_networks[network_id] = src
+
+	SSnetworks.networks[network_id] = src
+
+	SSnetworks.add_log("Network was created: [network_id]")
+
+	return ..()
+
+/datum/ntnet/vv_edit_var(var_name)
+	return FALSE // You cannot edit ANYTHING in here, don't even try you stupid admins!
```

Comment:
I will put a better description.  You change any of the vars in a net without changing the parent or the controlling devices things can go wonky.  I made the system dependent not on references to other datums but to just the network name string and the network_id string.  This way things can be destroyed and if a packet tried to reach it it would just fail.  I can plug in some helper functions that can change the network id, but any device that would broadcast on that network id wouldn't find any devices, etc.

 Still needs to be functions to do stuff like move devices to another network or rename it.  Those will be in part 2 as that will be adding an interface to change that in gametime.

Reply:
Generally even if vv is not supported directly we do not block it, that's reserved for preventing admin rights escalation and things that can crash the server directly.

Reply:
ah ok, in that case I will pull it.